Assets/MixedRealityToolkit.SDK/Experimental/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,343 @@
+        {
+            get => keepOrientationVertical;
+            set => keepOrientationVertical = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            " remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            " if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will 
+        /// remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal
+        /// if there is a hit."
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [Tooltip("This event is triggered once at the start of the placing state.")]
+        /// <summary>
+        /// This event is triggered once at the start of the placing state.
+        /// </summary>
+        public UnityEvent OnPlacingStarted = new UnityEvent();
+
+        [Tooltip("This event is triggered once when the placing state has ended.")]
+        /// <summary>
+        /// This event is triggered once when the placing state has ended.
+        /// </summary>
+        public UnityEvent OnPlacingStopped = new UnityEvent();
+
+        const int IgnoreRaycastLayer = 2;
+
+        // The current ray is based on the TrackedTargetType (Controller Ray, Head, Hand Joint)
+        protected RayStep currentRay;
+
+        protected bool didHitSurface;
+
+        protected RaycastHit currentHit;
+
+        protected float timeSinceClick = 0;
+
+        public int GameObjectLayer { get; protected set; }
+
+
+        #region MonoBehaviour Implementation
+        protected override void Start()
+        {
+            // Solver base class
+            base.Start();
+
+            if (!IsColliderPresent)
+            {
+                Debug.LogError("The GameObjectToPlace does not have a collider attached, please attach a collider");
+            }
+
+            SurfaceNormalOffset = gameObject.GetComponent<Collider>().bounds.extents.z;
+
+            // Store the initial game object layer
+            GameObjectLayer = gameObject.layer;
```

Comment:
this should be store in StartPlacement. Users could modify their layer after start but before placing

Reply:
Ah that's true, moved it into StartPlacement.

Reply:
I would also make sure to highlight this in the class remarks and in the documentation so people can easily be aware that the layer is being controlled by this component